src/components/PhotoSwipe/PhotoSwipe.tsx

```diff
@@ -528,6 +553,91 @@ function PhotoSwipe(props: Iprops) {
         }
     }, [showInfo]);
 
+    useEffect(() => {
+        if (!isOpen) return;
+        const item = items[photoSwipe?.getCurrentIndex()];
+        if (item && item.metadata.fileType === FILE_TYPE.LIVE_PHOTO) {
+            let videoPlaying = false;
+
+            const playVideo = async (livePhotoVideo, livePhotoImage) => {
+                if (videoPlaying) return;
+                videoPlaying = true;
+                livePhotoVideo.style.opacity = 1;
+                livePhotoImage.style.opacity = 0;
+                livePhotoVideo.load();
+                livePhotoVideo.play().catch(() => {
+                    pauseVideo(livePhotoVideo, livePhotoImage);
+                });
+            };
+
+            const pauseVideo = async (livePhotoVideo, livePhotoImage) => {
+                if (!videoPlaying) return;
+                videoPlaying = false;
+                livePhotoVideo.pause();
+                livePhotoVideo.style.opacity = 0;
+                livePhotoImage.style.opacity = 1;
+            };
+
+            const getvideoAndImage = () => {
+                const video = document.getElementsByClassName(
+                    `live-photo-video-${item.id}`
+                )[0];
+                const image = document.getElementsByClassName(
+                    `live-photo-image-${item.id}`
+                )[0];
+                return { video, image };
+            };
+
+            setLivePhotoBtnOptions({
+                click: () => {
+                    const { video, image } = getvideoAndImage();
```

Comment:
I was just thinking if we need to get the image and video element each time, or we could just get it once while initialising the click event and that would do ?

Reply:
Will try this.